### PR TITLE
php: Drop PHP 7.4

### PR DESCRIPTION
**Pull request by etu**
###### Description of changes
Drop PHP 7.4

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/php-packages.nix
Comment (jtojnar) [inline pkgs/top-level/php-packages.nix:565]: This is actually the opposite, we need to remove the xmlrpc extension.

Comment (etu) [reply pkgs/top-level/php-packages.nix:565]: Ah, right, pushed a fixup :slightly_smiling_face:

**Comment by etu** [comment]
1 package blacklisted:
nixos-install-tools

1 package failed to build:
arcanist

13 packages built:
cutelyst easyeffects engelsystem lsp-plugins php80Extensions.couchbase php80Extensions.ds php81Extensions.couchbase php81Extensions.ds pulseeffects-legacy snipe-it unit uwsgi virtlyst

**Review by yayayayaka — CHANGES_REQUESTED** [review]
> They state that they support PHP 8.1:
> https://snipe-it.readme.io/docs/requirements

This is not correct. (See [requirements for the v5.4.4 release](https://snipe-it.readme.io/v5.4.4/docs/requirements))

v6.0.1, which is ready for merge (https://github.com/NixOS/nixpkgs/pull/173272), does not require that constraint anymore, though.

**Comment by yayayayaka** [comment]
With #173272 merged, I have no other objections left.

**Comment by etu** [comment]
So it's now rebased on master :slightly_smiling_face:

**Comment by etu** [comment]
Similar result to before rebase:

2 packages blacklisted:
nixos-install-tools tests.nixos-functions.nixos-test

1 package failed to build:
arcanist

12 packages built:
cutelyst easyeffects engelsystem lsp-plugins php80Extensions.couchbase php80Extensions.ds php81Extensions.couchbase php81Extensions.ds pulseeffects-legacy unit uwsgi virtlyst

**Review by drupol — APPROVED** [review]
As discussed on Matrix: :+1:

File: pkgs/servers/http/unit/default.nix
Comment (drupol) [inline pkgs/servers/http/unit/default.nix:4]: Maybe this should have been replaced by `php81`, don't you think?

Comment (etu) [reply pkgs/servers/http/unit/default.nix:4]: Well, possibly, however I leave that up to the `unit` maintainers to decide what they want to support or not. I was just in there to not break evaluation :wink:

**Comment by jonringer** [comment]
Feel like dropping 7.4 completely was as little early, the 22.11 cycle doesn't branch off for another 3 months. 

Many downstream php projects are usually very slow to adopt a new version of php. Arcanist (which admittedly is no longer officially maintained) has [very little desire to fully fix php 8 compatibility](https://secure.phabricator.com/T13588)

**Comment by aanderse** [comment]
We're usually pretty aggressive on this. Too aggressive? Maybe not... We should add the `nix-phps` repository that @jtojnar (and to a much lesser extent a few others of us) maintain. This repository is a `php` developers dream come true and if properly documented in our nixpkgs manual I think this is ok. Keep in mind that upstream is pretty harsh about dropping support on an exact schedule, so I think our behavior was intended to reflect theirs.

If this rationale isn't convincing we can/should definitely start a conversation.

**Pull request merged**
